pkg/controller/service/servicecontroller.go

```diff
@@ -316,15 +312,40 @@ func (s *ServiceController) createLoadBalancerIfNeeded(namespacedName types.Name
 	} else {
 		glog.V(2).Infof("Ensuring LB for service %s", namespacedName)
 
-		// TODO: We could do a dry-run here if wanted to avoid the spurious cloud-calls & events when we restart
-
-		// The load balancer doesn't exist yet, so create it.
-		s.eventRecorder.Event(service, api.EventTypeNormal, "CreatingLoadBalancer", "Creating load balancer")
-		err := s.createLoadBalancer(service)
-		if err != nil {
-			return fmt.Errorf("Failed to create load balancer for service %s: %v", namespacedName, err), retryable
+		// Check if there is old LB there and if it need to be update
+		// Check if the balancer has already existed
+		// Avoid the spurious cloud-calls & events when we restart
+		skipCreate := false
+		if appliedState != nil {
+			// We should delete the old LB if the old LB does not
+			// match what we need now
```

Comment:
Hi, I make this modification since I think what our code doing is only create a new LB when we found that LB need and it is need to be updated. We did not make some special consideration of the old misconfigured LB. My code will delete the old misconfigured LB then create the new one, which I think will make it clean. I have read the code in your link, it does fixing misconfigured LB but may I know the reason why we leave the misconfigured LB here instead of deleting it?